src/tests/stream-tcp-list.c

```diff
@@ -545,15 +545,15 @@ static int StreamTcpReassembleTest01(void)
     OverlapBSDBefore(UINT_MAX - 10);
     OverlapBSDSame(UINT_MAX - 10);
     OverlapBSDAfter(UINT_MAX - 10);
-    return 1;
+    PASS;
```

Comment:
Considering that these `Overlap...After` functions all seem to call `OVERLAP_END`, which in term invokes `PASS`, I'm wondering - should we simply just remove the `return 1` line, to avoid extra calls? What is the best practice, in this case?

Reply:
For this one specifically, as the previous all is a function, not a macro we should end it with `PASS`, or `return OverLapBSDAfter(...)`;

But the function `StreamTcpReassembleTest01` is questionable.  If for example `OverlapBSDSame` fails, this unit test doesn't actually fail, so I think some larger refactoring needs to be done.

Looking at at `StreamTcpReassembleTest01`, the only place that unit test will actually fail out is the very first line.  Otherwise the test still passes even if the inner calls actually fail.  That is probably the confusion here as the tests look broken in the first place to me.

Reply:
Thanks... do you think we can split this work into two different tasks, or things are so mingled that we must refactor even before we ensure all comply with FAIL/PASS?

Reply:
Other than a few `return 1` this is mostly in compliance with FAIL/PASS.. I suppose you just change the `return 1;` to PASS and call it compliant.  Still broken, but compliant.

Essentially every call like `OverlapLINUXBefore(UINT_MAX - 10);` needs to be wrapped something like:
`FAIL_IF_NOT(OverlapLINUXBefore(UINT_MAX - 10));`.

Reply:
Thanks once more! Please, Liza, take Jason's comment's into account, and let us know if you have any questions, oke?